app/models/redis_notification.rb

```diff
@@ -45,10 +45,9 @@ def self.track(action, subject)
         current_user_id: User&.current&.id
       }
 
-      ['project_id', 'issue_id', 'user_id', 'member_id', 'role_id'].each do |attribute|
-        if subject.has_attribute?(attribute)
-          attributes[:additional_data] = Hash(attributes[:additional_data]).merge({attribute.to_sym => subject.send(attribute)})
-        end
+      attributes[:additional_data] = get_additional_data(subject, attributes[:additional_data])
+      if attributes[:additional_data] && attributes[:additional_data].key?(:member_id)
+        attributes[:additional_data] = Hash(attributes[:additional_data]).merge(get_additional_data(subject.member, attributes[:additional_data]))
```

Comment:
Some things:
* I find this method call a bit misleading, I'd rather call it `set_additional_data`.
* I'd move this entire additional_data definition in `set_additional_data` method or even include the additional_data setting in this method. These are class methods, it's weird having those mixed in the private section of this class.